returnn/tf/layers/basic.py

```diff
@@ -3414,35 +3406,47 @@ def get_out_data_from_opts(cls, name, axis, dims, pad_to_multiples=None, sources
     resolved_dims = None
     if dims and isinstance(dims[0], Dim):
       assert all(isinstance(d, Dim) for d in dims)
-      resolved_dims = dims
+      resolved_dims = tuple(dims)
       dims = [d.dimension or -1 for d in dims]
     if data.batch_shape[axis] is not None:
       resolved_shape_dims = cls._resolve_dims(
         old_dim=data.batch_shape[axis], new_dims=dims, pad_to_multiples=pad_to_multiples)
     else:
       resolved_shape_dims = tuple([(d if (d >= 0) else None) for d in dims])
 
-    rem_dim_indices = [i for i, d in enumerate(dims) if d < 0]
-    assert len(rem_dim_indices) <= 1, "only one entry in dims %r can be -1" % (dims,)
-    rem_dim_idx = rem_dim_indices[0] if rem_dim_indices else (len(dims) - 1)
-    axis_dim_tag = data.dim_tags[axis]
-    if all(d == 1 for (i, d) in enumerate(dims) if i != rem_dim_idx):
-      rem_dim = axis_dim_tag  # we can overtake the existing dim tag
-      if resolved_dims:
-        assert resolved_dims[rem_dim_idx] == axis_dim_tag
-    else:
-      rem_dim = Dim(
-        kind=axis_dim_tag.kind,
-        description="%s_split_dims%i_rem" % (name, rem_dim_idx),
-        dimension=resolved_shape_dims[rem_dim_idx])
     if not resolved_dims:
       resolved_dims = tuple(
         Dim(
-          kind=Dim.Types.Spatial,
+          kind=Dim.Types.Spatial if shape_dim is None else Dim.Types.Feature,
```

Comment:
The logic for `kind` is a bit incorrect or weird. I think there should not be multiple feature dims.

Why did you remove the `rem_dim` logic? I think you separately need to create a `rem_dim` which copies `axis_dim_tag.kind`, and here use it for `rem_dim_idx`.